### ISSUE TITLE
[Bug]: KeyError: 'c4TabbedHeaderRenderer' on specific youtube channel / playlist

### ISSUE THREAD

**Issue by jmincey** [tubearchivist/tubearchivist#467]
### I've read the documentation

- [X] I'm running the latest version of Tube Archivist and have read the [release notes](https://github.com/tubearchivist/tubearchivist/releases/latest).
- [X] I have read the [how to open an issue](https://github.com/tubearchivist/tubearchivist/blob/master/CONTRIBUTING.md#how-to-open-an-issue) guide, particularly the [bug report](https://github.com/tubearchivist/tubearchivist/blob/master/CONTRIBUTING.md#bug-report) section.

### Operating System

Synology

### Relevant log output

```shell
Ygj9vBF_WUk: change status to priority

[2023-04-19 22:10:47,344: INFO/MainProcess] Task download_pending[3c73ac1e-3ad4-41ec-ac36-5a6924f37ea9] received

[2023-04-19 22:10:47,346: WARNING/ForkPoolWorker-8] download_pending create callback

[2023-04-19 22:10:47,398: WARNING/ForkPoolWorker-8] Ygj9vBF_WUk: Downloading type: VideoTypeEnum.VIDEOS

[2023-04-19 22:11:21,188: WARNING/ForkPoolWorker-8] Ygj9vBF_WUk: get metadata from youtube

[2023-04-19 22:11:23,566: WARNING/ForkPoolWorker-8] UCtinbF-Q-fVthA0qrFQTgXQ: get metadata from es

[2023-04-19 22:11:25,734: WARNING/ForkPoolWorker-8] a4QufICobaA: Downloading type: VideoTypeEnum.VIDEOS

[2023-04-19 22:11:47,853: WARNING/ForkPoolWorker-8] a4QufICobaA: get metadata from youtube

[2023-04-19 22:11:50,277: WARNING/ForkPoolWorker-8] UCHF66aWLOxBW4l6VkSrS3cQ: get metadata from es

[2023-04-19 22:11:50,283: WARNING/ForkPoolWorker-8] {"_index":"ta_channel","_id":"UCHF66aWLOxBW4l6VkSrS3cQ","found":false}

[2023-04-19 22:11:50,283: WARNING/ForkPoolWorker-8] UCHF66aWLOxBW4l6VkSrS3cQ: scrape channel data from youtube

[2023-04-19 22:11:50,575: WARNING/ForkPoolWorker-8] 3c73ac1e-3ad4-41ec-ac36-5a6924f37ea9 Failed callback

[2023-04-19 22:11:50,577: ERROR/ForkPoolWorker-8] Task download_pending[3c73ac1e-3ad4-41ec-ac36-5a6924f37ea9] raised unexpected: KeyError('c4TabbedHeaderRenderer')

Traceback (most recent call last):

  File "/root/.local/lib/python3.10/site-packages/celery/app/trace.py", line 451, in trace_task

    R = retval = fun(*args, **kwargs)

  File "/root/.local/lib/python3.10/site-packages/celery/app/trace.py", line 734, in __protected_call__

    return self.run(*args, **kwargs)

  File "/app/home/tasks.py", line 193, in download_pending

    downloader.run_queue()

  File "/app/home/src/download/yt_dlp_handler.py", line 198, in run_queue

    vid_dict = index_new_video(

  File "/app/home/src/index/video.py", line 415, in index_new_video

    video.build_json()

  File "/app/home/src/index/video.py", line 151, in build_json

    self._add_channel()

  File "/app/home/src/index/video.py", line 207, in _add_channel

    channel.build_json(upload=True, fallback=self.youtube_meta)

  File "/app/home/src/index/channel.py", line 182, in build_json

    self.get_from_youtube(fallback)

  File "/app/home/src/index/channel.py", line 190, in get_from_youtube

    self.json_data = ChannelScraper(self.youtube_id).get_json()

  File "/app/home/src/index/channel.py", line 42, in get_json

    self._parse_channel_main()

  File "/app/home/src/index/channel.py", line 86, in _parse_channel_main

    main_tab = self.yt_json["header"]["c4TabbedHeaderRenderer"]

KeyError: 'c4TabbedHeaderRenderer'

Filter download view ignored only: True

Filter download view ignored only: False

https://www.youtube.com/playlist?list=PLIjqRbAQP0WKVbd1h9xb8gUPNCOxv02p2

[{'type': 'playlist', 'url': 'PLIjqRbAQP0WKVbd1h9xb8gUPNCOxv02p2', 'vid_type': 'unknown'}]

[2023-04-19 22:12:07,759: INFO/MainProcess] Task extract_download[ee9d5bda-23b5-4aa4-b163-caed97fdf0b0] received

[2023-04-19 22:12:07,761: WARNING/ForkPoolWorker-8] extract_download create callback

[2023-04-19 22:12:08,158: WARNING/ForkPoolWorker-8] PLIjqRbAQP0WKVbd1h9xb8gUPNCOxv02p2: get metadata from es

[2023-04-19 22:12:08,691: WARNING/ForkPoolWorker-8] UCHF66aWLOxBW4l6VkSrS3cQ: get metadata from es

[2023-04-19 22:12:08,697: WARNING/ForkPoolWorker-8] {"_index":"ta_channel","_id":"UCHF66aWLOxBW4l6VkSrS3cQ","found":false}

[2023-04-19 22:12:08,697: WARNING/ForkPoolWorker-8] UCHF66aWLOxBW4l6VkSrS3cQ: scrape channel data from youtube

[2023-04-19 22:12:09,097: WARNING/ForkPoolWorker-8] ee9d5bda-23b5-4aa4-b163-caed97fdf0b0 Failed callback

[2023-04-19 22:12:09,099: ERROR/ForkPoolWorker-8] Task extract_download[ee9d5bda-23b5-4aa4-b163-caed97fdf0b0] raised unexpected: KeyError('c4TabbedHeaderRenderer')

Traceback (most recent call last):

  File "/root/.local/lib/python3.10/site-packages/celery/app/trace.py", line 451, in trace_task

    R = retval = fun(*args, **kwargs)

  File "/root/.local/lib/python3.10/site-packages/celery/app/trace.py", line 734, in __protected_call__

    return self.run(*args, **kwargs)

  File "/app/home/tasks.py", line 201, in extrac_dl

    pending_handler.parse_url_list()

  File "/app/home/src/download/queue.py", line 182, in parse_url_list

    self._process_entry(entry)

  File "/app/home/src/download/queue.py", line 200, in _process_entry

    PlaylistSubscription().process_url_str([entry], subscribed=False)

  File "/app/home/src/download/subscriptions.py", line 197, in process_url_str

    self.channel_validate(playlist_h.json_data["playlist_channel_id"])

  File "/app/home/src/download/subscriptions.py", line 218, in channel_validate

    channel.build_json(upload=True)

  File "/app/home/src/index/channel.py", line 182, in build_json

    self.get_from_youtube(fallback)

  File "/app/home/src/index/channel.py", line 190, in get_from_youtube

    self.json_data = ChannelScraper(self.youtube_id).get_json()

  File "/app/home/src/index/channel.py", line 42, in get_json

    self._parse_channel_main()

  File "/app/home/src/index/channel.py", line 86, in _parse_channel_main

    main_tab = self.yt_json["header"]["c4TabbedHeaderRenderer"]

KeyError: 'c4TabbedHeaderRenderer'

[task][ee9d5bda-23b5-4aa4-b163-caed97fdf0b0]: received STOP signal.

[2023-04-19 22:12:14,340: INFO/MainProcess] Task download_pending[8815b32e-67ca-463a-b740-c13c1c077c67] received

[2023-04-19 22:12:14,342: WARNING/ForkPoolWorker-8] download_pending create callback

[2023-04-19 22:12:14,378: WARNING/ForkPoolWorker-8] download queue is empty

[2023-04-19 22:12:14,426: WARNING/ForkPoolWorker-8] gcrW53SoTKs: Downloading type: VideoTypeEnum.VIDEOS

[2023-04-19 22:12:16,554: INFO/MainProcess] Task update_subscribed[bc84ca26-be01-4683-aa79-496761dc8a9c] received

[2023-04-19 22:12:16,555: WARNING/ForkPoolWorker-1] update_subscribed create callback

[2023-04-19 22:12:16,761: WARNING/ForkPoolWorker-1] UCtinbF-Q-fVthA0qrFQTgXQ: find missing videos.

[2023-04-19 22:12:18,256: WARNING/ForkPoolWorker-1] ERROR: [youtube:tab] UCtinbF-Q-fVthA0qrFQTgXQ: This channel does not have a streams tab

[2023-04-19 22:12:18,256: WARNING/ForkPoolWorker-1] https://www.youtube.com/channel/UCtinbF-Q-fVthA0qrFQTgXQ/streams: failed to get info from youtube

[2023-04-19 22:12:18,897: WARNING/ForkPoolWorker-1] PL855S-vac7vdtXOLyJ5dJV51x6oaxb_Xg: get metadata from es

[2023-04-19 22:12:18,904: WARNING/ForkPoolWorker-1] PL855S-vac7vdtXOLyJ5dJV51x6oaxb_Xg: get metadata from youtube

[2023-04-19 22:12:20,387: WARNING/ForkPoolWorker-1] PLvcSNZqNYJCkOTGBs3HBLA4VfdL5X6yEw: get metadata from es

[2023-04-19 22:12:20,393: WARNING/ForkPoolWorker-1] PLvcSNZqNYJCkOTGBs3HBLA4VfdL5X6yEw: get metadata from youtube

[2023-04-19 22:12:22,185: WARNING/ForkPoolWorker-1] PLrwwpZzlar89l7al4rhaYyYhnbX3S9zxd: get metadata from es

[2023-04-19 22:12:22,191: WARNING/ForkPoolWorker-1] PLrwwpZzlar89l7al4rhaYyYhnbX3S9zxd: get metadata from youtube

[2023-04-19 22:12:23,385: WARNING/ForkPoolWorker-1] PLrwwpZzlar8_ohaKEfjlC6tRFmp-bSL1U: get metadata from es

[2023-04-19 22:12:23,391: WARNING/ForkPoolWorker-1] PLrwwpZzlar8_ohaKEfjlC6tRFmp-bSL1U: get metadata from youtube

[2023-04-19 22:12:24,594: WARNING/ForkPoolWorker-1] PLAzrgbu8gEMIXu-DKmffcoSDWA4tjbXPw: get metadata from es

[2023-04-19 22:12:24,600: WARNING/ForkPoolWorker-1] PLAzrgbu8gEMIXu-DKmffcoSDWA4tjbXPw: get metadata from youtube

[2023-04-19 22:12:25,836: WARNING/ForkPoolWorker-1] bc84ca26-be01-4683-aa79-496761dc8a9c success callback

[2023-04-19 22:12:25,838: INFO/ForkPoolWorker-1] Task update_subscribed[bc84ca26-be01-4683-aa79-496761dc8a9c] succeeded in 9.28096214600373s: None

[2023-04-19 22:12:34,728: WARNING/ForkPoolWorker-8] gcrW53SoTKs: get metadata from youtube

[2023-04-19 22:12:37,120: WARNING/ForkPoolWorker-8] UCHF66aWLOxBW4l6VkSrS3cQ: get metadata from es

[2023-04-19 22:12:37,125: WARNING/ForkPoolWorker-8] {"_index":"ta_channel","_id":"UCHF66aWLOxBW4l6VkSrS3cQ","found":false}

[2023-04-19 22:12:37,125: WARNING/ForkPoolWorker-8] UCHF66aWLOxBW4l6VkSrS3cQ: scrape channel data from youtube

[2023-04-19 22:12:37,442: WARNING/ForkPoolWorker-8] 8815b32e-67ca-463a-b740-c13c1c077c67 Failed callback

[2023-04-19 22:12:37,445: ERROR/ForkPoolWorker-8] Task download_pending[8815b32e-67ca-463a-b740-c13c1c077c67] raised unexpected: KeyError('c4TabbedHeaderRenderer')

Traceback (most recent call last):

  File "/root/.local/lib/python3.10/site-packages/celery/app/trace.py", line 451, in trace_task

    R = retval = fun(*args, **kwargs)

  File "/root/.local/lib/python3.10/site-packages/celery/app/trace.py", line 734, in __protected_call__

    return self.run(*args, **kwargs)

  File "/app/home/tasks.py", line 193, in download_pending

    downloader.run_queue()

  File "/app/home/src/download/yt_dlp_handler.py", line 198, in run_queue

    vid_dict = index_new_video(

  File "/app/home/src/index/video.py", line 415, in index_new_video

    video.build_json()

  File "/app/home/src/index/video.py", line 151, in build_json

    self._add_channel()

  File "/app/home/src/index/video.py", line 207, in _add_channel

    channel.build_json(upload=True, fallback=self.youtube_meta)

  File "/app/home/src/index/channel.py", line 182, in build_json

    self.get_from_youtube(fallback)

  File "/app/home/src/index/channel.py", line 190, in get_from_youtube

    self.json_data = ChannelScraper(self.youtube_id).get_json()

  File "/app/home/src/index/channel.py", line 42, in get_json

    self._parse_channel_main()

  File "/app/home/src/index/channel.py", line 86, in _parse_channel_main

    main_tab = self.yt_json["header"]["c4TabbedHeaderRenderer"]

KeyError: 'c4TabbedHeaderRenderer'

Wed Apr 19 22:12:58 2023 - SIGPIPE: writing to a closed pipe/socket/fd (probably the client disconnected) on request /static/favicon/favicon-16x16.de290f6207d0.png (ip 172.17.0.1) !!!

Wed Apr 19 22:12:58 2023 - uwsgi_response_sendfile_do(): Broken pipe [core/writer.c line 655] during GET /static/favicon/favicon-16x16.de290f6207d0.png (172.17.0.1)

OSError: write error

https://www.youtube.com/watch?v=HoU2kkaf_nI

[{'type': 'video', 'url': 'HoU2kkaf_nI', 'vid_type': 'unknown'}]

[2023-04-19 22:14:18,145: INFO/MainProcess] Task extract_download[441a4677-97c4-4fc3-ace0-7738cfa70221] received

[2023-04-19 22:14:18,146: WARNING/ForkPoolWorker-8] extract_download create callback

[2023-04-19 22:14:18,343: WARNING/ForkPoolWorker-8] HoU2kkaf_nI: [1/1]: add to queue

[2023-04-19 22:14:20,539: WARNING/ForkPoolWorker-8] 441a4677-97c4-4fc3-ace0-7738cfa70221 success callback

[2023-04-19 22:14:20,542: INFO/ForkPoolWorker-8] Task extract_download[441a4677-97c4-4fc3-ace0-7738cfa70221] succeeded in 2.3932942079845816s: None

HoU2kkaf_nI: change status to priority

[2023-04-19 22:14:27,650: INFO/MainProcess] Task download_pending[fd4a447f-a28c-4ca2-b93b-833568231ad1] received

[2023-04-19 22:14:27,651: WARNING/ForkPoolWorker-8] download_pending create callback

[2023-04-19 22:14:27,704: WARNING/ForkPoolWorker-8] HoU2kkaf_nI: Downloading type: VideoTypeEnum.VIDEOS

[2023-04-19 22:15:00,113: WARNING/ForkPoolWorker-8] HoU2kkaf_nI: get metadata from youtube

[2023-04-19 22:15:02,478: WARNING/ForkPoolWorker-8] UCHF66aWLOxBW4l6VkSrS3cQ: get metadata from es

[2023-04-19 22:15:02,483: WARNING/ForkPoolWorker-8] {"_index":"ta_channel","_id":"UCHF66aWLOxBW4l6VkSrS3cQ","found":false}

[2023-04-19 22:15:02,484: WARNING/ForkPoolWorker-8] UCHF66aWLOxBW4l6VkSrS3cQ: scrape channel data from youtube

[2023-04-19 22:15:02,773: WARNING/ForkPoolWorker-8] fd4a447f-a28c-4ca2-b93b-833568231ad1 Failed callback

[2023-04-19 22:15:02,775: ERROR/ForkPoolWorker-8] Task download_pending[fd4a447f-a28c-4ca2-b93b-833568231ad1] raised unexpected: KeyError('c4TabbedHeaderRenderer')

Traceback (most recent call last):

  File "/root/.local/lib/python3.10/site-packages/celery/app/trace.py", line 451, in trace_task

    R = retval = fun(*args, **kwargs)

  File "/root/.local/lib/python3.10/site-packages/celery/app/trace.py", line 734, in __protected_call__

    return self.run(*args, **kwargs)

  File "/app/home/tasks.py", line 193, in download_pending

    downloader.run_queue()

  File "/app/home/src/download/yt_dlp_handler.py", line 198, in run_queue

    vid_dict = index_new_video(

  File "/app/home/src/index/video.py", line 415, in index_new_video

    video.build_json()

  File "/app/home/src/index/video.py", line 151, in build_json

    self._add_channel()

  File "/app/home/src/index/video.py", line 207, in _add_channel

    channel.build_json(upload=True, fallback=self.youtube_meta)

  File "/app/home/src/index/channel.py", line 182, in build_json

    self.get_from_youtube(fallback)

  File "/app/home/src/index/channel.py", line 190, in get_from_youtube

    self.json_data = ChannelScraper(self.youtube_id).get_json()

  File "/app/home/src/index/channel.py", line 42, in get_json

    self._parse_channel_main()

  File "/app/home/src/index/channel.py", line 86, in _parse_channel_main

    main_tab = self.yt_json["header"]["c4TabbedHeaderRenderer"]

KeyError: 'c4TabbedHeaderRenderer'
```

### Anything else?

Logs above show a video downloading as expected (watch?v=Ygj9vBF_WUk)
and the failing video (watch?v=HoU2kkaf_nI) 
I am using the latest release,  TubeArchivist v0.3.5 and all other containers are on the latest versions. 

redis/redis-stack-server:latest
bbilly1/tubearchivist-es:latest
bbilly1/tubearchivist:latest

**Comment by bbilly1** [comment]
Excellent, thanks for collecting this here!

**Comment by bbilly1** [comment]
Thanks for reporting, that is now fixed in v0.3.6, please update.
